org.xaraxtreme.XaraLX.json

```diff
@@ -0,0 +1,779 @@
+                "install -d '/app/share/doc/imagemagick/Magick++'; install -p -m 0644 'Magick++/AUTHORS' 'Magick++/ChangeLog' 'Magick++/NEWS' 'Magick++/README' '/app/share/doc/imagemagick/Magick++/';",
+                "install -d '/app/share/doc/imagemagick/Magick++/examples'; install -p -m 0644 Magick++/examples/* '/app/share/doc/imagemagick/Magick++/examples/';",
+                "echo 'post-install: licenses';",
+                "install -d '/app/share/licenses/imagemagick'; install -p -m 0644 'LICENSE' '/app/share/licenses/imagemagick/';",
+                "install -d '/app/share/licenses/imagemagick/Magick++'; install -p -m 0644 'www/Magick++/COPYING' '/app/share/licenses/imagemagick/Magick++/';"
+            ]
+        },
+        {
+            "name": "wxGTK2",
+            "no-autogen": true,
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://downloads.sf.net/wxwindows/wxGTK-2.8.12.tar.bz2",
+                    "sha256": "f4193c29fb0e790c9a5c8936f082377a7e51c76bccafe41f4f9da7ca15c0ef1a"
+                },
+                {
+                    "type": "file",
+                    "path": "wx-config",
+                    "sha256": "9002d9b69d0ff9a1b67cae59fc61783a1fd16a6f2213bc79af60ff1ee364ed1c"
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'fix wx-config';",
+                        "sed -r -i 's!lib64!lib!g;s!/usr/!/app/!g' 'wx-config';"
+                    ]
+                },
+                {
+                    "type": "patch",
+                    "path": "wxGTK-2.8.12-test.patch",
+                    "strip-components": 1,
+                    "sha256": "217ed70bacd95b3be51d462155d87cecf27a789fca9fd7a9334f4a4f586e6b02"
+                },
+                {
+                    "type": "patch",
+                    "path": "wxGTK-2.8.12-abicheck.patch",
+                    "strip-components": 1,
+                    "sha256": "46a1bb97d69163547da13d5e23a4c73e68de27ee601da5d2fb5bc5c417931453"
+                },
+                {
+                    "type": "patch",
+                    "path": "wxGTK-2.8.12-char.patch",
+                    "strip-components": 1,
+                    "sha256": "5025cf8a73405106c8c55c819a92006088da53ed0256e8e41b4c58cd93d354be"
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'fix permissions for sources';",
+                        "chmod a-x include/wx/{msgout.h,dcgraph.h,graphics.h} src/common/msgout.cpp;"
+                    ]
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'configure';",
```

Comment:
This should be rewritten to use autotools buildsystem.

Reply:
How to handle multiple targets with just pure **autotools** buildsystem?

https://github.com/scx/flathub/blob/a2559882da5b47071a5627f7c517339d9f1f5962/org.xaraxtreme.XaraLX.json#L266

Anyway, the "custom" buildsystem (**simple** or pseudo-**autotools**) is used by many flatpak packages in the flathub repo, e.g. `com.anydesk.Anydesk`, `com.github.alecaddd.sequeler`, `com.github.ojubaorg.Othman`, `com.lettier.movie-monad`, `com.play0ad.zeroad`, `io.github.cloose.CuteMarkEd`, `io.github.jliljebl.Flowblade`, `io.qt.qtwebkit.BaseApp`, `net.ankiweb.Anki.json`, `net.minetest.Minetest`, `net.openra.OpenRA`, `net.pcsx2.PCSX2`, `net.sourceforge.torcs`, `org.blender.Blender`, `org.flatpak.Builder` (sic!), `org.freedesktop.Bustle`, `org.freedesktop.Sdk.Extension.openjdk9`, `org.freefilesync.FreeFileSync`, `org.gnome.Eolie`, `org.gnucash.GnuCash`, `org.gottcode.FocusWriter`, `org.gpodder.gpodder`, `org.kde.kdenlive`, `org.kde.okular`, `org.libreoffice.LibreOffice`, `org.mozilla.Thunderbird`, `org.musescore.MuseScore`, `org.octave.Octave`, `org.openshot.OpenShot`, `org.shotcut.Shotcut`, `org.subsurface_divelog.Subsurface`, `org.telegram.desktop`, `org.tuxpaint.Tuxpaint`, `cld2`.

http://pastebin.centos.org/766716/13011152/

Reply:
I don't want to spend time discussing this. Almost all mentioned applications use `simple` build system for things that should use it: when build tool is completely foreign to flatpak-builder or there is no build system involved at all.

Reply:
It doesn't matter anymore, but as far as I know this is still impossible using autotools.
https://github.com/scx/xaralx-flatpak/blob/c3e6906136ac9d7085e5d732743e1287e694fc82/wxGTK2.yaml#L51-L61